docs/development/design/automation-rules.rst

```diff
@@ -0,0 +1,101 @@
+Automation Rules
+================
+
+Automation rules allow users to apply some actions over new or deleted versions.
+See :doc:`/automation-rules` for more details of the current features.
+
+This document explore some ideas to expand automation rules to automate more tasks.
+
+Current state and limitations
+-----------------------------
+
+- Matches are done using regular expressions.
+  But the models were designed so other type of matches can be implemented
+  (like comparing the version with the highest semver version).
+- Automation rules are meant to automate existing actions,
+  so an action shouldn't be able to be triggered only inside an automation rule.
+  This is so users aren't blocked if a rule didn't match a version,
+  or if they need to execute the action over existing versions.
+- Automation rules are currently matched against new versions,
+  but they can also be executed over deleted versions (https://github.com/readthedocs/readthedocs.org/pull/7644).
+
+Improvements
+------------
+
+Keep track of the ``n`` latest matches
```

Comment:
This is something that I've wanting for some time. I think it's super useful to keep track of what happened to understand why my versions have changed. Also, to debug mistakes on my own rules.